Assets/_Laz/Scripts/Environment/ElectricGateBehaviour.cs

```diff
@@ -34,22 +37,42 @@ public class ElectricGateBehaviour : MonoBehaviour
 
         private ElectricGate _gate;
 
-        public void Initialize()
+        private bool _isActive = true;
+
+        public override void Initialize()
         {
-            _gate = new ElectricGate(_onTime, _offTime, _flickering);
-            _gate.OnGateFlickerChange += HandleGateStateChange;
+            //Only called by puzzle manager. If this isn't set up with a puzzle, Initialize won't be called
+            Reset();
         }
 
-        public void CleanUp()
+        public override void CleanUp()
         {
-            _gate.CleanUp();
```

Comment:
do u need to do a base.CleanUp()?

Reply:
Don't forget to test the reset of the game. "Press R" to reset the game to check if it works a second round through